code/game/objects/objs.dm

```diff
@@ -63,6 +66,85 @@
 		visible_message("<span class='danger'>[src] shatters into a million pieces!</span>")
 		qdel(src)
 
+/obj/proc/start_pulling(atom/movable/AM,gs)

```

Comment:
copy paste makes me sad.

Reply:
Yup, it's pulling mechanics from mob. Do you have better ideas?

Reply:
I can just put it in atom/moveable to avoid copypasting.

Reply:
Yes, moving it to movable (their common parent) is best.

Reply:
I was told on coderbus,that I should just leave it in obj. But okay.

Reply:
By who? Common parent functionality is infinitely better than duplicated functionality across two types.

Reply:
Can't remember.maybe AnturK, but not sure. Also there is a problem,because it was modifyed. and not fully consistent with mob.

Reply:
Even where not consistent, you can move the procs themselves to atom/movable, and simply override them to provide the unique parts of the functionality.